src/main/java/com/netgrif/application/engine/pdf/generator/service/fieldbuilder/FieldBuilder.java

```diff
@@ -148,6 +141,10 @@ private boolean isDgFlow(DataGroup dataGroup) {
         return dataGroup.getLayout() != null && dataGroup.getLayout().getType() != null && dataGroup.getLayout().getType().equals("flow");
     }
 
+    private boolean isDgLegacy(DataGroup dataGroup) {
+        return dataGroup.getLayout() == null || dataGroup.getLayout().getType() == null || dataGroup.getLayout().getType().equals("legacy");
```

Comment:
it would be better if both both of these layout string constants were real constants and not string literals

Reply:
Done